greenriver/hmis-frontend
# Support custom client detail forms (for pathways)

## Description

Add a "Custom Fields" section to the client dashboard. It gets populated if there are any active forms with role `CLIENT_DETAIL`. This can be used to collect or display arbitrary data on the client dashboard.

The card is basically the same as the `EnrollmentDetails` card on the Enrollment Dashboard, so I did some refactoring to allow them to share code.

Editing these fields requires `can_edit_clients` perm for the client. Viewing these fields requires `can_view_clients` perm for the client.

<img width="902" alt="Screenshot 2024-01-10 at 3 55 00 PM" src="https://github.com/greenriver/hmis-frontend/assets/5333982/325f42b0-dafe-454d-9879-8f6758d406d6">


### How to test

see instructions on backend pr

[//]: # 'remove if not applicable'
Depends on hmis-warehouse PR: https://github.com/greenriver/hmis-warehouse/pull/3898

## Type of change

## Checklist before requesting review
